### DISCUSSION TITLE
[Enhancement] Please prioritize minimizing application startup speed

### DISCUSSION THREAD

**Discussion by Happypig375** [dotnet/maui#75] [General] [answered]
This is a long-standing issue of Xamarin.Forms - we all saw the native apps that start in less than 1 second which cannot ever be achieved by Xamarin.Forms. Please prioritize minimizing application startup speed!

**Comment by JohnH-FF** [comment]
Do you mean startup time on Android? I definitely agree, they are appalling, especially on legacy devices.
But our XF apps on iOS, MacOS and UWP all start up sub second, thankfully.

**Comment by Happypig375** [comment]
Yes

**Comment by JonDouglas** [comment]
@Happypig375 We are consistently delivering startup speed performance on platforms like Android. For example, a recent feature to bring startup speed from 4.8s -> 1.9s without a huge APK size bloat.

https://devblogs.microsoft.com/xamarin/faster-android-startup-times-with-startup-tracing/

We will continue to invest in these areas as we are making significant progress to that 1s milestone.

**Reply by trampster** [reply]
We use profiled startup tracing in our Xamarin.Forms Shell app, and startup is 3 seconds. We also do compiled xaml, have delayed all our app init until after the UI is shown. We use a custom IoC container which doesn't use reflection, but it is still horribly slow.

**Comment by PureWeen** [comment, marked as answer]
@Happypig375 this is a big focus the handlers are going to let us really thin down the layouts, we're reworking the layout system, getting rid of all assembly scanning, and whatever else we can find. 

We're adding a lot of instrumentation to maintain a perspective on performance as we build MAUI up

**Reply by Happypig375** [reply]
Does this mean that the Xamarin Profiler will be free for all?

**Comment by microspaze** [comment]
It seems that the maui app startup speed is always slow than 5 seconds. Is there any way to reduce it? It's too too too slow for demo app to show the basic UI.

**Reply by JohnH-FF** [reply]
Release builds are best for demo, debug builds are not optimised for startup.

**Comment by azrinsani** [comment]
MAUI is approaching release, any update on this? Personally this is the ONLY reason for me to switch to Flutter. The startup time for Android and (IOS as well to some extent) is unbearable. It should be below 1 second on a standard mid tier phone.
